backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h

```diff
@@ -4237,6 +4241,842 @@ template <typename Torus> struct int_comparison_buffer {
+  int_sub_and_propagate<Torus> *sub_and_propagate_mem;
+  int_bitop_buffer<Torus> *bitor_mem_1;
+  int_bitop_buffer<Torus> *bitor_mem_2;
+  int_bitop_buffer<Torus> *bitor_mem_3;
+  int_logical_scalar_shift_buffer<Torus> *shift_mem;
+
+  // lookup tables
+  int_radix_lut<Torus> *message_extract_lut_1;
+  int_radix_lut<Torus> *message_extract_lut_2;
+  int_radix_lut<Torus> *zero_out_if_not_1_lut_1;
+  int_radix_lut<Torus> *zero_out_if_not_1_lut_2;
+  int_radix_lut<Torus> *zero_out_if_not_2_lut_1;
+  int_radix_lut<Torus> *zero_out_if_not_2_lut_2;
+  int_radix_lut<Torus> *quotient_lut_1;
+  int_radix_lut<Torus> *quotient_lut_2;
+  int_radix_lut<Torus> *quotient_lut_3;
+
+  // sub streams
+  CudaStreams sub_streams_1;
+
+  // temporary device buffers
+  CudaRadixCiphertextFFI *d1;                  // num_blocks + 1
+  CudaRadixCiphertextFFI *d2;                  // num_blocks + 1
+  CudaRadixCiphertextFFI *d3;                  // num_blocks + 1
+  CudaRadixCiphertextFFI *tmp_gpu_0;           // num_blocks + 1
+  CudaRadixCiphertextFFI *tmp_gpu_1;           // num_blocks + 1
+  CudaRadixCiphertextFFI *tmp_gpu_2;           // num_blocks + 1
+  CudaRadixCiphertextFFI *tmp_gpu_3;           // num_blocks + 1
+  CudaRadixCiphertextFFI *divisor_gpu_1;       // num_blocks
+  CudaRadixCiphertextFFI *divisor_gpu_2;       // num_blocks
+  CudaRadixCiphertextFFI *remainder_gpu_1;     // num_blocks
+  CudaRadixCiphertextFFI *remainder_gpu_2;     // num_blocks
+  CudaRadixCiphertextFFI *remainder_gpu_3;     // num_blocks
+  CudaRadixCiphertextFFI *low1;                // num_blocks
+  CudaRadixCiphertextFFI *low2;                // num_blocks
+  CudaRadixCiphertextFFI *low3;                // num_blocks
+  CudaRadixCiphertextFFI *rem0;                // num_blocks
+  CudaRadixCiphertextFFI *rem1;                // num_blocks
+  CudaRadixCiphertextFFI *rem2;                // num_blocks
+  CudaRadixCiphertextFFI *rem3;                // num_blocks
+  CudaRadixCiphertextFFI *sub_result_1;        // num_blocks
+  CudaRadixCiphertextFFI *sub_result_2;        // num_blocks
+  CudaRadixCiphertextFFI *sub_result_3;        // num_blocks
+  CudaRadixCiphertextFFI *sub_1_overflowed;    // num_blocks
+  CudaRadixCiphertextFFI *sub_2_overflowed;    // num_blocks
+  CudaRadixCiphertextFFI *sub_3_overflowed;    // num_blocks
+  CudaRadixCiphertextFFI *comparison_blocks_1; // num_blocks
+  CudaRadixCiphertextFFI *comparison_blocks_2; // num_blocks
+  CudaRadixCiphertextFFI *comparison_blocks_3; // num_blocks
+  CudaRadixCiphertextFFI *cmp_1;               // boolean block
+  CudaRadixCiphertextFFI *cmp_2;               // boolean block
+  CudaRadixCiphertextFFI *cmp_3;               // boolean block
+  CudaRadixCiphertextFFI *c0;                  // single block
+  CudaRadixCiphertextFFI *q1;                  // single block
+  CudaRadixCiphertextFFI *q2;                  // single block
+  CudaRadixCiphertextFFI *q3;                  // single block
```

Comment:
Looking at this list I'm wondering if we couldn't avoid allocating so much data by reusing some of it, but for now i's fine.

Reply:
compred to previous implementation I tried to reuse some memory that was obvious that it could be reuse, there are probably more to be reused, maybe later we can do.